### DISCUSSION TITLE
how can i achieve early initialization of beans like spring @configuration

### DISCUSSION THREAD

**Discussion by sdyava** [micronaut-projects/micronaut-core#6445] [Q&A] [answered]
i'm new to micronaut, how can i achieve what spring achieves with @configuration. Here is the sample code. I tried @Context and @singleton, but with no luck.

@Configuration
@EnableBinding(Sink.class)
@Slf4j
@Profile("default")
@Component
public class StreamConfig {
	
	@Autowired
	private Environment env;
    
	
	@Bean(name = "sslproperties")
	public Properties entityManagerFactory() {

       
        System.setProperty("javax.net.ssl.trustStore", env.getProperty("spring.kafka.ssl.trust-store-location").replaceAll("file://", "").replaceAll("file:", ""));
        System.setProperty("javax.net.ssl.trustStorePassword", env.getProperty("spring.kafka.ssl.trust-store-password"));
        System.setProperty("javax.net.ssl.keyStore", env.getProperty("spring.kafka.ssl.key-store-location").replaceAll("file://", "").replaceAll("file:", ""));
        System.setProperty("javax.net.ssl.keyStorePassword", env.getProperty("spring.kafka.ssl.key-password"));
        System.setProperty("javax.net.ssl.trustStoreType", "jks");

		return System.getProperties();
	}

**Comment by graemerocher** [comment]
More information is required, what didn't work for you with `@Context`?

**Comment by sdyava** [comment]
Here is what i'm trying with @Context . I just want to initialize system variables on the server start up. please help me on what i'm missing here.

@Context
public class ConsumerStreamConfig {
    @Value("trust-store-location")
    String trustStore;
    @Value("trust-store.password")
    String trustStorePwd;
    @Value("key-store-location")
    String keyStore;
    @Value("key-store.password")
    String keyStorePwd;


	@Context
	public Properties entityManagerFactory() {

        System.setProperty("javax.net.ssl.trustStore", trustStore);
        System.setProperty("javax.net.ssl.trustStorePassword", trustStorePwd);
        System.setProperty("javax.net.ssl.keyStore", keyStore);
        System.setProperty("javax.net.ssl.keyStorePassword", keyStorePwd);
        System.setProperty("javax.net.ssl.trustStoreType", "jks");




		return System.getProperties();
	}
}

**Reply by graemerocher** [reply]
`ConsumerStreamConfig` needs to be annotated with `@Factory` and the `entityManagerFactory` method with `@Context`

**Reply by sdyava** [reply]
Thank you for answering . Cant i refer @Value here. trying to inject below values from application.yaml
@Factory
public class ConsumerStreamConfig {
    @Value("kafka.ssl.truststore.location")
    String trustStore;
    @Value("kafka.ssl.truststore.password")
    String trustStorePwd;
    @Value("kafka.ssl.keystore.location")
    String keyStore;
    @Value("kafka.ssl.keystore.password")
    String keyStorePwd;


application.yaml

kafka:
  ssl:
 
    keystore:
      #location: ${KEYSTORE_LOCATION:/app/certs/prod/keystore.jks}
      location: ${KEYSTORE_LOCATION:src/main/resources/certs/dev/keystore.jks}
      password: ${KEYSTORE_PASSWORD:******}
    truststore:
      location: ${TRUSTSTORE_LOCATION:src/main/resources/certs/dev/truststore.jks}
      password: ${TRUSTSTORE_PASSWORD:****}

**Reply by sdyava** [reply]
had to ${"kafka.ssl.truststore.location"}, this worked.

**Comment by sdyava** [comment, marked as answer]
sorry i meant creating bean like this 

@Bean
public Properties entityManagerFactory() {

    System.setProperty("javax.net.ssl.trustStore", trustStore);
    System.setProperty("javax.net.ssl.trustStorePassword", trustStorePwd);
    System.setProperty("javax.net.ssl.keyStore", keyStore);
    System.setProperty("javax.net.ssl.keyStorePassword", keyStorePwd);
    System.setProperty("javax.net.ssl.trustStoreType", "jks");




	return System.getProperties();
}
